test/correctness/simd_op_check.cpp

```diff
@@ -511,6 +511,17 @@ class SimdOpCheck : public SimdOpCheckTest {
             check("vcvtneps2bf16*ymm", 16, cast(BFloat(16), f32_1));
             check("vcvtneps2bf16*xmm", 8, cast(BFloat(16), f32_1));
             check("vcvtneps2bf16*xmm", 4, cast(BFloat(16), f32_1));
+
+            {
+                // 16 bit, 2 element dot product
+                RDom r(0, 2);
+                check("vdpbf16ps*zmm", 16, sum(f32(in_bf16(2 * x + r)) * in_bf16(2 * x + r + 32)));
+                check("vdpbf16ps*ymm", 8, sum(f32(in_bf16(2 * x + r)) * in_bf16(2 * x + r + 32)));
+                check("vdpbf16ps*xmm", 4, sum(f32(in_bf16(2 * x + r)) * in_bf16(2 * x + r + 32)));
+                // We match on `sum(f32 * f32)` for bf16 dot product, so check
+                // that we correctly generate code for the f32 dot product too.
+                check("", 4, sum(in_f32(2 * x + r) * in_f32(2 * x + r + 32)));
```

Comment:
This string is empty? I'm confused.

Reply:
I don't mind what instructions get generated for the f32 dot product, but want to be sure that it will generate something that behaves correctly. This test catches the potential error if I had not [changed the user_assert](https://github.com/halide/Halide/pull/5712/files#diff-6cd3cc01186f9ce4abaa1a98c9201bd440983ecc5620fbccae383df1116bdce6L507-R518). Is there a better way of checking this?

Reply:
I think https://github.com/halide/Halide/blob/master/test/correctness/vector_reductions.cpp will already test this, so there's no need to add anything to simd_op_check.